aws/resource_aws_dx_lag.go

```diff
@@ -0,0 +1,162 @@
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDxLagBandWidth,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"num_connections": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"force_destroy": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceAwsDxLagCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.CreateLagInput{
+		ConnectionsBandwidth: aws.String(d.Get("band_width").(string)),
+		LagName:              aws.String(d.Get("name").(string)),
+		Location:             aws.String(d.Get("location").(string)),
+		NumberOfConnections:  aws.Int64(int64(d.Get("num_connections").(int))),
+	}
+	resp, err := conn.CreateLag(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.LagId)
+	return resourceAwsDxLagRead(d, meta)
+}
+
+func resourceAwsDxLagRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	lagId := d.Id()
+	input := &directconnect.DescribeLagsInput{
+		LagId: aws.String(lagId),
+	}
+	resp, err := conn.DescribeLags(input)
+	if err != nil {
+		d.SetId("")
+		return err
+	}
+	if len(resp.Lags) != 1 {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] Number of DX Lag (%s) isn't one, got %d", lagId, len(resp.Lags))
+	}
+	if d.Id() != *resp.Lags[0].LagId {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] DX Lag (%s) not found", lagId)
+	}
+	return nil
+}
+
+func resourceAwsDxLagUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.UpdateLagInput{
+		LagId: aws.String(d.Id()),
+	}
+	if d.HasChange("name") {
+		input.LagName = aws.String(d.Get("name").(string))
+	}
+	_, err := conn.UpdateLag(input)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceAwsDxLagDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	if d.Get("force_destroy").(bool) {
+		input := &directconnect.DescribeLagsInput{
+			LagId: aws.String(d.Id()),
+		}
+		resp, err := conn.DescribeLags(input)
+		if err != nil {
+			return err
+		}
+		lag := resp.Lags[0]
+		for _, v := range lag.Connections {
+			dcinput := &directconnect.DeleteConnectionInput{
+				ConnectionId: v.ConnectionId,
+			}
+			if _, err := conn.DeleteConnection(dcinput); err != nil {
+				return err
+			}
+		}
+	}
+
+	input := &directconnect.DeleteLagInput{
+		LagId: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteLag(input)
+	if err != nil {
+		return err
+	}
+	deleteStateConf := &resource.StateChangeConf{
+		Pending:    []string{directconnect.LagStateAvailable, directconnect.LagStateRequested, directconnect.LagStatePending},
+		Target:     []string{directconnect.LagStateDeleted, directconnect.LagStateDeleting},
```

Comment:
Is there any reason we should treat `directconnect.LagStateDeleting` as `Target` state here?